test/tests_basic.jl

```diff
@@ -44,6 +44,56 @@ function refine_fn_balance(p4est, which_tree, quadrant)
   return Cint(0)
 end
 
+# This function belongs to the testset "nested attributes" below
+# first, some auxiliary functions
+function unsafe_load_sc(::Type{T}, sc_array::Ptr{sc_array}, i=1) where T
+  sc_array_obj = unsafe_load(sc_array)
+  return unsafe_load_sc(T, sc_array_obj, i)
+end
+
+function unsafe_load_sc(::Type{T}, sc_array_obj::sc_array, i=1) where T
+  element_size = sc_array_obj.elem_size
+  @assert element_size == sizeof(T)
+
+  return unsafe_load(Ptr{T}(sc_array_obj.array), i)
+end
+
+function unsafe_load_side(info::Ptr{p4est_iter_face_info_t}, i=1)
+  return unsafe_load_sc(p4est_iter_face_side_t, unsafe_load(info).sides, i)
+end
+
+function iter_face_nested_attributes(info::Ptr{p4est_iter_face_info_t}, user_data)
+  ptr = Ptr{Bool}(user_data)
+  # TODO: How to test if error occures on C side?
+  no_test_fail = unsafe_load(ptr, 1)
+  if @test_nowarn unsafe_load(info).sides.elem_count == 2
```

Comment:
My idea was that we test here whether accesses like these provide the correct types and not some pointers to make sure we didn't miss an `unsafe_load` or so. Thus, this could be

```suggestion
  @test unsafe_load(info).sides.elem_count isa Integer
  if unsafe_load(info).sides.elem_count == 2
```
The other tests should be adapted similarly. Sorry for the unclear communication.

Reply:
Ok, I'll adapt the tests.